### PR TITLE
[cli] Honor `--local-config` when present during `vc build`

### PR DESCRIPTION
**Pull request by cb1kenobi**
As reported https://github.com/vercel/vercel/discussions/9648, `vc build` does not honor the `--local-config <file>` option.

`vc build` will only load the `vercel.json` (or `now.json`) in the `workPath` which is based on the `rootDirctory`.

If `--local-config` is specified in the command line arguments, then it should take precedence.

### CODE REVIEW

File: packages/cli/test/integration.test.ts [hunk 1]
```diff
@@ -3683,3 +3683,41 @@ test('vercel.json configuration overrides in an existing project do not prompt u
   text = await page.text();
   expect(text).toMatch(/Next\.js Test/);
 });
+
+test('[vc build] should use --local-config over default vercel.json', async () => {
```
Comment (TooTallNate) [inline packages/cli/test/integration.test.ts]: It'd prefer to add this as a unit test instead.

File: packages/cli/src/index.ts [hunk 2]
```diff
@@ -267,6 +267,7 @@ const main = async () => {
     config,
     authConfig,
     localConfig,
+    localConfigPath,
```
Comment (TooTallNate) [inline packages/cli/src/index.ts:270]: Seems like we could just use `localConfig`, instead of passing in a new prop for the path?

**Review by TooTallNate — CHANGES_REQUESTED** [review]

**Comment by cb1kenobi** [comment]
Fixes #9670

File: packages/cli/test/integration.test.ts (hunk 1 above)
Comment (cb1kenobi) [reply packages/cli/test/integration.test.ts]: Fixed in 8fac37560bfadce326a36bed1563da3196350856

**Review by styfle — APPROVED** [review]

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**